File: fix-images.js
import fs from "fs";

const filePath = "data/products.ts";
let content = fs.readFileSync(filePath, "utf-8");

const picsumUrls = Array.from({ length: 50 }, (_, i) => `https://picsum.photos/800/800?random=${i + 1}`);
let counter = 0;

// Replace all unsplash URLs
content = content.replace(/https:\/\/images\.unsplash\.com\/photo-[^?]+\?[^"]+/g, () => {
  const url = picsumUrls[counter % picsumUrls.length];
  counter++;
  return url;
});

content = content.replace(/https:\/\/plus\.unsplash\.com\/photo-[^?]+\?[^"]+/g, () => {
  const url = picsumUrls[counter % picsumUrls.length];
  counter++;
  return url;
});

fs.writeFileSync(filePath, content, 'utf-8');
console.log('✓ Successfully replaced all Unsplash URLs with picsum.photos');

File: fix_images.py
import re

# Read the file
with open('data/products.ts', 'r', encoding='utf-8') as f:
    content = f.read()

# List of reliable picsum URLs
picsum_urls = [f'https://picsum.photos/800/800?random={i}' for i in range(1, 50)]

# Replace all unsplash URLs
counter = 0
def replace_url(match):
    global counter
    url = picsum_urls[counter % len(picsum_urls)]
    counter += 1
    return url

# Replace all variations of unsplash URLs
pattern1 = r'https://images\.unsplash\.com/photo-[^?]+\?[^"]*'
pattern2 = r'https://plus\.unsplash\.com/photo-[^?]+\?[^"]*'

content = re.sub(pattern1, replace_url, content)

# Reset counter for second pass
counter = 0
content = re.sub(pattern2, replace_url, content)

# Write back
with open('data/products.ts', 'w', encoding='utf-8') as f:
    f.write(content)

print('✓ Successfully replaced all Unsplash URLs with picsum.photos')
